trade-finance-manager-api/src/v1/controllers/user/user.controller.js

```diff
@@ -31,120 +59,101 @@ exports.findByUsername = async (username, callback) => {
-      // remove the raw password
-      delete userUpdate.password;
-      delete userUpdate.passwordConfirm;
-      delete userUpdate.currentPassword;
-
-      // create new salt/hash for the new password
-      const { salt, hash } = utils.genPassword(newPassword);
-      // queue update of salt+hash, ie store the encrypted password
-      userUpdate.salt = salt;
-      userUpdate.hash = hash;
-      // queue the addition of the old salt/hash to our list of blocked passwords that we re-check
-      // in 'passwordsCannotBeReUsed' rule
-      userUpdate.blockedPasswordList = oldBlockedPasswordList.concat([{ oldSalt, oldHash }]);
+    if (!ObjectId.isValid(_id)) {
+      throw new Error('Error Updating TFM user - Invalid User Id');
     }
 
+    const userUpdate = {
+      ...update,
+      auditDetails: sessionUser?._id ? generateTfmUserAuditDetails(sessionUser._id) : generateNoUserLoggedInAuditDetails(),
+    };
+
+    const collection = await db.getCollection('tfm-users');
+
     await collection.updateOne({ _id: { $eq: ObjectId(_id) } }, { $set: userUpdate }, {});
 
-    callback(null, userUpdate);
-  });
-};
+    callback();
+  } catch (error) {
+    console.error('Error Updating TFM user %s', error);
 
-exports.updateLastLoginAndResetSignInData = async (user, sessionIdentifier, callback) => {
-  if (!ObjectId.isValid(user._id)) {
-    throw new Error('Invalid User Id');
+    throw new Error('Error Updating TFM user %s', error);
   }
+};
 
-  const collection = await db.getCollection('tfm-users');
-  const update = {
-    lastLogin: Date.now(),
-    loginFailureCount: 0,
-    sessionIdentifier,
-    auditDetails: generateTfmUserAuditDetails(user._id),
-  };
-  await collection.updateOne({ _id: { $eq: ObjectId(user._id) } }, { $set: update }, {});
+/**
+ * updateLastLoginAndResetSignInData
+ * Update a user's "last login" and reset sign in data.
+ * @param {Object} user
+ * @param {String} sessionIdentifier
+ * @param {Function} callback: Callback function. defaults to an empty function.
+ * @returns {Promise<Function>} Callback function
+ */
+exports.updateLastLoginAndResetSignInData = async (user, sessionIdentifier, callback = () => { }) => {
+  try {
+    console.info('Updating TFM user - last login, reset sign in data');
 
-  callback();
-};
+    if (!ObjectId.isValid(user._id)) {
+      throw new Error('Error Updating TFM user - last login, reset sign in data - Invalid User Id');
+    }
 
-exports.incrementFailedLoginCount = async (user) => {
-  if (!ObjectId.isValid(user._id)) {
-    throw new Error('Invalid User Id');
-  }
+    const collection = await db.getCollection('tfm-users');
 
-  const failureCount = user.loginFailureCount ? user.loginFailureCount + 1 : 1;
-  const thresholdReached = failureCount >= businessRules.loginFailureCount;
+    const update = {
+      lastLogin: Date.now(),
+      sessionIdentifier,
+      auditDetails: generateTfmUserAuditDetails(user._id),
```

Comment:
Really odd this is showing up here -- is this out of date with main

Reply:
SSO branch is updated from time to time. 

I did another update now, it replaced `auditDetails` with `auditRecord`.